Client.Core/Flux/Internal/FluxResultMapper.cs

```diff
@@ -30,53 +35,52 @@ internal T ToPoco<T>(FluxRecord record)
             try
             {
                 var type = typeof(T);
-                var poco = (T) Activator.CreateInstance(type);
+                var poco = (T)Activator.CreateInstance(type);
+
+                // copy record to case insensitive dictionary (do this once)
+                var recordValues =
+                    new Dictionary<string, object>(record.Values, StringComparer.InvariantCultureIgnoreCase);
 
-                var properties = type.GetProperties();
+                var properties = PropertyCache.GetOrAdd(type, _ => type.GetProperties());
 
                 foreach (var property in properties)
                 {
-                    var attributes = property.GetCustomAttributes(typeof(Column), false);
-
-                    Column attribute = null;
-
-                    if (attributes.Length > 0)
-                    {
-                        attribute = (Column) attributes.First();
-                    }
-
-                    var columnName = property.Name;
-
-                    if (attribute != null && !string.IsNullOrEmpty(attribute.Name))
-                    {
-                        columnName = attribute.Name;
-                    }
-
-                    // copy record to case insensitive dictionary
-                    var recordValues =
-                        new Dictionary<string, object>(StringComparer.InvariantCultureIgnoreCase);
-
-                    foreach (var entry in record.Values)
+                    Column attribute = AttributeCache.GetOrAdd(property, _ =>
                     {
-                        recordValues.Add(entry.Key, entry.Value);
-                    }
-
-                    string col = null;
+                        var attributes = property.GetCustomAttributes(typeof(Column), false);
+                        return attributes.Length > 0 ? attributes[0] as Column : null;
+                    });
 
-                    if (recordValues.ContainsKey(columnName))
+                    if (attribute != null && attribute.IsTimestamp)
                     {
-                        col = columnName;
+                        SetFieldValue(poco, property, record.GetTimeInDateTime());
```

Comment:
We should also support a timestamp field with the `Instant` type. The POCO could looks like:

```csharp
private class PointWithoutTimestampNameInstant
{
    [Column("tag", IsTag = true)] public string Tag { get; set; }
    [Column("min")] public double Minimum { get; set; }
    [Column("max")] public double Maximum { get; set; }
    [Column("avg")] public Double Average { get; set; }
    [Column(IsTimestamp = true)] public Instant Timestamp { get; set; }
}
```

Could you please add a following test into `FluxResultMapperTest.cs`:

```csharp
[Test]
public void MapTimestampWithInstantTime()
{
    var now = DateTime.UtcNow;
    
    var fluxRecord = new FluxRecord(0);
    fluxRecord.Values["tag"] = "production";
    fluxRecord.Values["min"] = 10.5;
    fluxRecord.Values["max"] = 20.0;
    fluxRecord.Values["avg"] = (Double) 18;
    fluxRecord.Values["_time"] = Instant.FromDateTimeUtc(now);
    
    var poco = _parser.ToPoco<PointWithoutTimestampNameInstant>(fluxRecord);
    
    Assert.AreEqual("production", poco.Tag);
    Assert.AreEqual(10.5, poco.Minimum);
    Assert.AreEqual(20, poco.Maximum);
    Assert.AreEqual(18, poco.Average);
    Assert.AreEqual(Instant.FromDateTimeUtc(now), poco.Timestamp);
}

private class PointWithoutTimestampNameInstant
{
    [Column("tag", IsTag = true)] public string Tag { get; set; }
    [Column("min")] public double Minimum { get; set; }
    [Column("max")] public double Maximum { get; set; }
    [Column("avg")] public Double Average { get; set; }
}   
```